src/bucket/BucketSnapshotManager.h

```diff
@@ -48,6 +49,11 @@ class BucketSnapshotManager : NonMovableOrCopyable
     // Lock must be held when accessing any member variables holding snapshots
     mutable SharedMutex mSnapshotMutex;
 
+    mutable Mutex mSimpleTimerRegistryMutex;
```

Comment:
I don't think it makes much sense for BucketSnapshotManager to manage the metrics registry. While this initial PR just deals with BucketList related loads, there are other time based metrics that also require simple timer, see "herder.pending[-soroban]-txs.sum". I think what makes the most sense to me is to create a lightweight wrapper that combines our SimpleTimer metric registry with `medida::MetricsRegistry`. Maybe we could call this something like `stellar::MetricRegistry` and have this be the entry point for both registries. This type would then be returned by `Application::getMetrics`.

Reply:
Ya I think given the sync requirement and in order to maintain a unified interface with our other metrics, it would be helpful.